### PR TITLE
fix(rest): maintain trailing slash to resource parameter

### PR DESCRIPTION
**Pull request by doktordirk**
eg:
apiEndpoint.find('posts/') -> path = http://endpoint/posts/
apiEndpoint.find('posts/', 'id')) -> path = http://endpoint/posts/id/
apiEndpoint.find('posts/', {user:1}) -> path = http://endpoint/posts/?user=1

closing https://github.com/SpoonX/aurelia-api/issues/140

---
This change is [Reviewable](https://reviewable.io/reviews/spoonx/aurelia-api/141)

### CODE REVIEW

**Comment by doktordirk** [comment]
@RWOverdijk comments?

**Comment by CoinBR** [comment]
Great Job =)  
That works great for me!
Thank you ;D

**Comment by RWOverdijk** [comment]
I prefer something like this:

```js
let normalized = (original.replace(/^\/|\/$/g, '')) + '/';
```

It's easier, less code and more readable.

**Comment by doktordirk** [comment]
though that means that there.ll be always a trailing slash. i.m nit sure that there won't be scenario's where that's not desired

**Comment by RWOverdijk** [comment]
I meant to add between the hostname and path. Anything after, I don't know. If you need a trailing slash sure, we can just leave it there. But imo all endpoints should work without it.

**Comment by doktordirk** [comment]
@RWOverdijk      https://github.com/SpoonX/aurelia-api/issues/140

**Comment by RWOverdijk** [comment]
My bad. In that case I'd make it a config option, or use . I don't like adding a trailing space to the resource name. Using regex for this every single time is just too much. Besides, this would be a breaking change and I don't feel like we should do a major bump for this.

If it's a config option you have my +1.

File: src/rest.js [hunk 1]
```diff
@@ -136,7 +136,14 @@ export class Rest {
 }
 
 function getRequestPath(resource, criteria) {
-  return (criteria !== undefined && criteria !== null
-    ? resource + (typeof criteria !== 'object' ? `/${criteria}` : '?' + buildQueryString(criteria))
-    : resource);
+  if (typeof criteria === 'object' && criteria !== null) {
+    resource += `?${buildQueryString(criteria)}`;
+  } else if (criteria) {
+    if (resource.slice(-1) === '/') {
```
Comment (doktordirk) [inline src/rest.js]: the only change

**Comment by CoinBR** [comment]
I'm not in a position to advocate a certain approach, but this is what in question:

> In Unix, a trailing slash on a pathname identifies the path as pointing to a folder (aka directory). If a pathname does not have a trailing slash then it points to a file. A folder is a ‘collection’ of files.

The syntax of URIs is derived from the syntax of Unix filenames, and the concept of using trailing slashes to identify ‘collection’ resources was carried over. However on the Web, the strong delineation between folders and files does not exist, frequently a ‘collection’ resource appears similar in structure and content to a normal resource (sometimes referred to as a ‘subordinate’ resource).

As a consequence, much confusion has arisen about the purpose and importance of trailing slashes on collection resources. It is common for users to forget the trailing slash on a resource, and **common for web-servers to assist users, when they make this mistake, to redirect them to the URI with the trailing slash automatically.**

[Source](https://cdivilly.wordpress.com/2014/03/11/why-trailing-slashes-on-uris-are-important/)

In my case, the api is considering everything as a resource, and always redirecting my requests to a url with a trailing slash.

**Comment by doktordirk** [comment]
no in so many words, but that's what i wanted to say here https://github.com/SpoonX/aurelia-api/pull/141#issuecomment-236524550 
a trailing slash is usually required, but usually one adds that in .htaccess if missing

**Pull request merged**